aten/src/ATen/core/SingletonSymNodeImpl.h

```diff
@@ -131,17 +167,38 @@ class TORCH_API SingletonSymNodeImpl : public SymNodeImpl {
   c10::SymNode mul(const c10::SymNode& other) override;
 
   c10::optional<int64_t> singleton_int() override {
+    TORCH_CHECK(
+        type_ == SingletonVariant::PYTHON,
+        "shape returned from strided layout NestedTensor does not support this "
+        "operation");
     return val_;
   }
 
   c10::optional<int64_t> singleton_coeff() override {
+    TORCH_INTERNAL_ASSERT(type_ == SingletonVariant::PYTHON);
     return coeff_;
   }
 
+  // If we would like to have singleton_vec() as a virtual method, it must
+  // be defined on SymNodeImpl, which exists in c10 only. This means we cannot
+  // return normal Tensor. The workaround here is to return a pointer instead.
+  // Instead of using this method directly, please use get_singleton_vec, if you
+  // need a regular Tensor.
+  c10::TensorImpl* singleton_vec() const override {
```

Comment:
This signature means it's impossible to return an owning reference to TensorImpl ever. (Not needed here though.)